bridge/eth/client.go

```diff
@@ -61,30 +60,24 @@ type Client struct {
 	gasHandler       bridge.GasHandler
 }
 
-func NewClient(
-	config bridge.EthereumConfig,
-	broadcaster bridge.Broadcaster,
-	mapper bridge.Mapper,
-	gasHandler bridge.GasHandler,
-) (*Client, error) {
+// ArgsClient is the DTO used in the client constructor
+type ArgsClient struct {
+	Config      bridge.EthereumConfig
+	Broadcaster bridge.Broadcaster
+	Mapper      bridge.Mapper
+	GasHandler  bridge.GasHandler
+	EthClient   BlockchainClient
+	EthInstance BridgeContract
+}
+
+func NewClient(args ArgsClient) (*Client, error) {
 
 	log := logger.GetOrCreate("EthClient")
 
-	if check.IfNil(gasHandler) {
+	if check.IfNil(args.GasHandler) {
```

Comment:
there are no nil checks on other args fields: Config, Broadcaster, Mapper etc. 
same thing also in other files that use args

Reply:
Yes, indeed we need that. But we can proceed with this one like it is and 1 follow-up PR will come with changes for all missed check in all args types.

Reply:
[Check args for all args types #105](https://github.com/ElrondNetwork/elrond-eth-bridge/pull/105)